yjit_codegen.c

```diff
@@ -118,6 +118,18 @@ jit_peek_at_self(jitstate_t *jit, ctx_t *ctx)
     return jit->ec->cfp->self;
 }
 
+static VALUE
+jit_peek_at_local(jitstate_t *jit, ctx_t *ctx, int n)
+{
+    RUBY_ASSERT(jit_at_current_insn(jit));
+
+    int32_t local_table_size = jit->iseq->body->local_table_size;
+    RUBY_ASSERT(n < (int)jit->iseq->body->local_table_size);
+
+    const VALUE *ep = jit->ec->cfp->ep;
+    return ep[-VM_ENV_DATA_SIZE - local_table_size + n + 1];
+}
+
 // When we know a VALUE to be static, this returns an appropriate val_type_t
 static val_type_t
 jit_type_of_value(VALUE val)
```

Comment:
I think this function would probably belong in `yjit_core.c`, but it seems like a very useful addition.

Reply:
👍 Was beginning to think that. Thanks for the confirmation